### ISSUE TITLE
Clipping

### ISSUE THREAD

**Issue by sjc9999** [helix-toolkit/helix-toolkit#1795]
![ConeSphere](https://user-images.githubusercontent.com/23642863/179013725-9d656772-8090-42e3-8655-d08412349544.jpg)

As you see in the picture I have a cone with a base and a sphere.
I would like the cone not to draw anywhere it is not visible from point A.
So the hole in the base of the cone should be the diameter of the sphere.
So that it I was looking from the base of the cone to point A none of the sphere would appear to be covered by the cone.
Kind of like what a spot light would do if it was at point a pointing at the sphere.

**Comment by sjc9999** [comment]
No I'm trying to make the hole in the base the same size as the diameter of the sphere is this particular case.  But in general I want my cone not to be drawn if a point on the cone is not visible from a observer at point A.

Here are some other views:
![ConeSphere2](https://user-images.githubusercontent.com/23642863/179079417-958c6556-a4bc-4b1e-862a-731a1551ceb0.jpg)

![ConeSphere3](https://user-images.githubusercontent.com/23642863/179079463-07e61ce4-c508-4b4f-b28f-3148ecf27be8.jpg)

Here is what I want it to look like from the back verse the image above:


![ConeSphere4](https://user-images.githubusercontent.com/23642863/179079559-1910e945-04b0-4f55-9573-a3e771574e5b.jpg)

**Comment by holance** [comment]
What happens if you rotates the view. Is the ball still clipping the cone shape?

**Comment by sjc9999** [comment]
Yes, but if the scene is rotate such that the camera is looking straight from A to the center of the sphere you can't tell any clipping is happening is this example.
What I'm showing here is the simplest example some cones will still intersect the sphere but might be off center from the sphere.

**Comment by holance** [comment]
Probably you can just calculate the cone and sphere intersection diameter and length to the diameter, then generate the cone mesh?

**Comment by holance** [comment]
Feels like the `TopMostGroup3D` is the one you are looking for. Just put the sphere model inside the `TopMostGroup3D`.

**Comment by sjc9999** [comment]
Nope TopMostGroup3D does not have the desired affect.  It always puts the sphere over the cone.
I think the custom shader is the right approach.

What are the shaders (vertex and pixel) being used for a MeshGeometryModel3D with only DiffuseMaterial DiffuseColor set?

**Comment by holance** [comment]
Probably you can use two sphere, one in most top group, one is outside. Then you can enable the one in top most when camera moves to the point you specified.
